// @medusajs/medusa/dist/api-v2/admin/inventory-items/validators.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminPostInventoryItemsInventoryItemParams = exports.AdminPostInventoryItemsInventoryItemReq = exports.AdminPostInventoryItemsItemLocationLevelsLevelParams = exports.AdminPostInventoryItemsItemLocationLevelsLevelReq = exports.AdminPostInventoryItemsReq = exports.AdminPostInventoryItemsItemLocationLevelsParams = exports.AdminPostInventoryItemsItemLocationLevelsReq = exports.AdminGetInventoryItemsParams = exports.AdminGetInventoryItemsItemParams = void 0;
var common_1 = require("../../../types/common");
var class_validator_1 = require("class-validator");
var class_transformer_1 = require("class-transformer");
var utils_1 = require("../../../utils");
var AdminGetInventoryItemsItemParams = /** @class */ (function (_super) {
    __extends(AdminGetInventoryItemsItemParams, _super);
    function AdminGetInventoryItemsItemParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminGetInventoryItemsItemParams;
}(common_1.FindParams));
exports.AdminGetInventoryItemsItemParams = AdminGetInventoryItemsItemParams;
/**
 * Parameters used to filter and configure the pagination of the retrieved inventory items.
 */
var AdminGetInventoryItemsParams = /** @class */ (function (_super) {
    __extends(AdminGetInventoryItemsParams, _super);
    function AdminGetInventoryItemsParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([String, [String]]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "id", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminGetInventoryItemsParams.prototype, "q", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([String, [String]]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "location_id", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([String, [String]]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "sku", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([String, [String]]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "origin_country", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([String, [String]]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "mid_code", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([String, [String]]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "material", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([String, [String], common_1.StringComparisonOperator]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "hs_code", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([Number, common_1.NumericalComparisonOperator]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "weight", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([Number, common_1.NumericalComparisonOperator]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "length", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([Number, common_1.NumericalComparisonOperator]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "height", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, utils_1.IsType)([Number, common_1.NumericalComparisonOperator]),
        __metadata("design:type", Object)
    ], AdminGetInventoryItemsParams.prototype, "width", void 0);
    __decorate([
        (0, class_validator_1.IsBoolean)(),
        (0, class_validator_1.IsOptional)(),
        (0, class_transformer_1.Transform)(function (_a) {
            var value = _a.value;
            return value === "true";
        }),
        __metadata("design:type", Boolean)
    ], AdminGetInventoryItemsParams.prototype, "requires_shipping", void 0);
    return AdminGetInventoryItemsParams;
}((0, common_1.extendedFindParamsMixin)({
    limit: 20,
    offset: 0,
})));
exports.AdminGetInventoryItemsParams = AdminGetInventoryItemsParams;
var AdminPostInventoryItemsItemLocationLevelsReq = /** @class */ (function () {
    function AdminPostInventoryItemsItemLocationLevelsReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsItemLocationLevelsReq.prototype, "location_id", void 0);
    __decorate([
        (0, class_validator_1.IsNumber)(),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsItemLocationLevelsReq.prototype, "stocked_quantity", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsNumber)(),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsItemLocationLevelsReq.prototype, "incoming_quantity", void 0);
    return AdminPostInventoryItemsItemLocationLevelsReq;
}());
exports.AdminPostInventoryItemsItemLocationLevelsReq = AdminPostInventoryItemsItemLocationLevelsReq;
// eslint-disable-next-line
var AdminPostInventoryItemsItemLocationLevelsParams = /** @class */ (function (_super) {
    __extends(AdminPostInventoryItemsItemLocationLevelsParams, _super);
    function AdminPostInventoryItemsItemLocationLevelsParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminPostInventoryItemsItemLocationLevelsParams;
}(common_1.FindParams));
exports.AdminPostInventoryItemsItemLocationLevelsParams = AdminPostInventoryItemsItemLocationLevelsParams;
/**
 * @schema AdminPostInventoryItemsReq
 * type: object
 * description: "The details of the inventory item to create."
 * properties:
 *   sku:
 *     description: The unique SKU of the associated Product Variant.
 *     type: string
 *   ean:
 *     description: The EAN number of the item.
 *     type: string
 *   upc:
 *     description: The UPC number of the item.
 *     type: string
 *   barcode:
 *     description: A generic GTIN field for the Product Variant.
 *     type: string
 *   hs_code:
 *     description: The Harmonized System code of the Inventory Item. May be used by Fulfillment Providers to pass customs information to shipping carriers.
 *     type: string
 *   inventory_quantity:
 *     description: The amount of stock kept of the associated Product Variant.
 *     type: integer
 *     default: 0
 *   allow_backorder:
 *     description: Whether the associated Product Variant can be purchased when out of stock.
 *     type: boolean
 *   manage_inventory:
 *     description: Whether Medusa should keep track of the inventory for the associated Product Variant.
 *     type: boolean
 *     default: true
 *   weight:
 *     description: The weight of the Inventory Item. May be used in shipping rate calculations.
 *     type: number
 *   length:
 *     description: The length of the Inventory Item. May be used in shipping rate calculations.
 *     type: number
 *   height:
 *     description: The height of the Inventory Item. May be used in shipping rate calculations.
 *     type: number
 *   width:
 *     description: The width of the Inventory Item. May be used in shipping rate calculations.
 *     type: number
 *   origin_country:
 *     description: The country in which the Inventory Item was produced. May be used by Fulfillment Providers to pass customs information to shipping carriers.
 *     type: string
 *   mid_code:
 *     description: The Manufacturers Identification code that identifies the manufacturer of the Inventory Item. May be used by Fulfillment Providers to pass customs information to shipping carriers.
 *     type: string
 *   material:
 *     description: The material and composition that the Inventory Item is made of, May be used by Fulfillment Providers to pass customs information to shipping carriers.
 *     type: string
 *   title:
 *     description: The inventory item's title.
 *     type: string
 *   description:
 *     description: The inventory item's description.
 *     type: string
 *   thumbnail:
 *     description: The inventory item's thumbnail.
 *     type: string
 *   metadata:
 *     description: An optional set of key-value pairs with additional information.
 *     type: object
 *     externalDocs:
 *       description: "Learn about the metadata attribute, and how to delete and update it."
 *       url: "https://docs.medusajs.com/development/entities/overview#metadata-attribute"
 */
var AdminPostInventoryItemsReq = /** @class */ (function () {
    function AdminPostInventoryItemsReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsReq.prototype, "sku", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsReq.prototype, "hs_code", void 0);
    __decorate([
        (0, class_validator_1.IsNumber)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsReq.prototype, "weight", void 0);
    __decorate([
        (0, class_validator_1.IsNumber)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsReq.prototype, "length", void 0);
    __decorate([
        (0, class_validator_1.IsNumber)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsReq.prototype, "height", void 0);
    __decorate([
        (0, class_validator_1.IsNumber)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsReq.prototype, "width", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsReq.prototype, "origin_country", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsReq.prototype, "mid_code", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsReq.prototype, "material", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsReq.prototype, "title", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsReq.prototype, "description", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsReq.prototype, "thumbnail", void 0);
    __decorate([
        (0, class_validator_1.IsObject)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Object)
    ], AdminPostInventoryItemsReq.prototype, "metadata", void 0);
    return AdminPostInventoryItemsReq;
}());
exports.AdminPostInventoryItemsReq = AdminPostInventoryItemsReq;
/**
 * @schema AdminPostInventoryItemsItemLocationLevelsLevelReq
 * type: object
 * properties:
 *   stocked_quantity:
 *     description: the total stock quantity of an inventory item at the given location ID
 *     type: number
 *   incoming_quantity:
 *     description: the incoming stock quantity of an inventory item at the given location ID
 *     type: number
 */
var AdminPostInventoryItemsItemLocationLevelsLevelReq = /** @class */ (function () {
    function AdminPostInventoryItemsItemLocationLevelsLevelReq() {
    }
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsNumber)(),
        (0, class_validator_1.Min)(0),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsItemLocationLevelsLevelReq.prototype, "incoming_quantity", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsNumber)(),
        (0, class_validator_1.Min)(0),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsItemLocationLevelsLevelReq.prototype, "stocked_quantity", void 0);
    return AdminPostInventoryItemsItemLocationLevelsLevelReq;
}());
exports.AdminPostInventoryItemsItemLocationLevelsLevelReq = AdminPostInventoryItemsItemLocationLevelsLevelReq;
// eslint-disable-next-line
var AdminPostInventoryItemsItemLocationLevelsLevelParams = /** @class */ (function (_super) {
    __extends(AdminPostInventoryItemsItemLocationLevelsLevelParams, _super);
    function AdminPostInventoryItemsItemLocationLevelsLevelParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminPostInventoryItemsItemLocationLevelsLevelParams;
}(common_1.FindParams));
exports.AdminPostInventoryItemsItemLocationLevelsLevelParams = AdminPostInventoryItemsItemLocationLevelsLevelParams;
/**
 * @schema AdminPostInventoryItemsInventoryItemReq
 * type: object
 * description: "The attributes to update in an inventory item."
 * properties:
 *   hs_code:
 *     description: The Harmonized System code of the Inventory Item. May be used by Fulfillment Providers to pass customs information to shipping carriers.
 *     type: string
 *   origin_country:
 *     description: The country in which the Inventory Item was produced. May be used by Fulfillment Providers to pass customs information to shipping carriers.
 *     type: string
 *   mid_code:
 *     description: The Manufacturers Identification code that identifies the manufacturer of the Inventory Item. May be used by Fulfillment Providers to pass customs information to shipping carriers.
 *     type: string
 *   material:
 *     description: The material and composition that the Inventory Item is made of, May be used by Fulfillment Providers to pass customs information to shipping carriers.
 *     type: string
 *   weight:
 *     description: The weight of the Inventory Item. May be used in shipping rate calculations.
 *     type: number
 *   height:
 *     description: The height of the Inventory Item. May be used in shipping rate calculations.
 *     type: number
 *   width:
 *     description: The width of the Inventory Item. May be used in shipping rate calculations.
 *     type: number
 *   length:
 *     description: The length of the Inventory Item. May be used in shipping rate calculations.
 *     type: number
 *   title:
 *     description: The inventory item's title.
 *     type: string
 *   description:
 *     description: The inventory item's description.
 *     type: string
 *   thumbnail:
 *     description: The inventory item's thumbnail.
 *     type: string
 *   requires_shipping:
 *     description: Whether the item requires shipping.
 *     type: boolean
 */
var AdminPostInventoryItemsInventoryItemReq = /** @class */ (function () {
    function AdminPostInventoryItemsInventoryItemReq() {
    }
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "sku", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "origin_country", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "hs_code", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "mid_code", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsString)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "material", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsNumber)(),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "weight", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsNumber)(),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "height", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsNumber)(),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "length", void 0);
    __decorate([
        (0, class_validator_1.IsOptional)(),
        (0, class_validator_1.IsNumber)(),
        __metadata("design:type", Number)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "width", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "title", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "description", void 0);
    __decorate([
        (0, class_validator_1.IsString)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", String)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "thumbnail", void 0);
    __decorate([
        (0, class_validator_1.IsBoolean)(),
        (0, class_validator_1.IsOptional)(),
        __metadata("design:type", Boolean)
    ], AdminPostInventoryItemsInventoryItemReq.prototype, "requires_shipping", void 0);
    return AdminPostInventoryItemsInventoryItemReq;
}());
exports.AdminPostInventoryItemsInventoryItemReq = AdminPostInventoryItemsInventoryItemReq;
var AdminPostInventoryItemsInventoryItemParams = /** @class */ (function (_super) {
    __extends(AdminPostInventoryItemsInventoryItemParams, _super);
    function AdminPostInventoryItemsInventoryItemParams() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return AdminPostInventoryItemsInventoryItemParams;
}(common_1.FindParams));
exports.AdminPostInventoryItemsInventoryItemParams = AdminPostInventoryItemsInventoryItemParams;
